sitl_launcher/launch/demo.launch.py

```diff
@@ -0,0 +1,126 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import launch
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from ament_index_python.packages import get_package_share_directory
+from launch.actions import SetEnvironmentVariable
+import sys
+from launch import LaunchService
+from launch_ros import get_default_launch_description
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from launch.conditions import IfCondition
+from launch.event_handlers import OnProcessExit
+from launch import LaunchContext
+from launch.utilities import perform_substitutions, normalize_to_list_of_substitutions
+
+from launch.actions import IncludeLaunchDescription
+from launch import LaunchDescriptionSource
+from launch.substitutions import PythonExpression
+from typing import Dict
+from typing import List
+from typing import Text
+
+class ReplaceWorldString(launch.Substitution):
+  """
+  Substitution that replaces strings on a given file.
+  """
+
+  def __init__(self,
+    sitl_world: launch.SomeSubstitutionsType) -> None:
+    super().__init__()
+
+    from launch.utilities import normalize_to_list_of_substitutions  # import here to avoid loop
+    self.__sitl_world = normalize_to_list_of_substitutions(sitl_world)
+
+  @property
+  def name(self) -> List[launch.Substitution]:
+    """Getter for name."""
+    return self.__sitl_world
+
+  def perform(self, context: launch.LaunchContext) -> Text:
+
+      sitl_world = launch.utilities.perform_substitutions(context, self.name)
+
+      if(sitl_world=="yosemite"):
+          pkg_shared_directory = 'yosemite_valley'
+          launch_file = 'drone_yosemite.launch.py'
+      elif(sitl_world=="mcmillan"):
+          pkg_shared_directory = 'mcmillan_airfield'
+          launch_file = 'drone_mcmillan.launch.py'
+      elif(sitl_world=="ksql"):
+          pkg_shared_directory = 'ksql_airport'
+          launch_file = 'drone_ksql.launch.py'
+      elif(sitl_world=="baylands"):
+          pkg_shared_directory = 'baylands'
+          launch_file = 'drone_baylands.launch.py'
+
+      world_dir = get_package_share_directory(pkg_shared_directory)
+      return world_dir + '/launch/' + launch_file
+
+
+def generate_launch_description():
+
+    declare_sitl_world_cmd = DeclareLaunchArgument(
+        'sitl_world',
+        default_value='yosemite',
+        description='World [yosemite, mcmillan, ksql, baylands]')
+
+    sitl_world = ReplaceWorldString(sitl_world=launch.substitutions.LaunchConfiguration('sitl_world'))
+
+    included_launch = launch.actions.IncludeLaunchDescription(
+        launch.launch_description_sources.PythonLaunchDescriptionSource(
+                sitl_world))
+
+    use_rviz = LaunchConfiguration('use_rviz')
+    declare_use_rviz_cmd = DeclareLaunchArgument(
+        'use_rviz',
```

Comment:
When I launch this with True I get two copies of rviz, one with the config and one plain. 

And when it's false I just get the plain rviz. 

![Screenshot from 2019-11-13 15-25-53](https://user-images.githubusercontent.com/447804/68813246-1b061d00-062a-11ea-9176-d1021d629e4f.png)

Reply:
@tfoote can you verify your entrypoint.sh ?

Reply:
The one with the config file up it's the rviz launched by the launch file but I included something in the entrypoint to launch an rviz instance but I removed it https://github.com/osrf/drone_demo/commit/373fb54242631849e78867ae12e2db3ad18298e5#diff-b958f585a04af5ee2087610ea7180f34